src/pip/_internal/utils/temp_dir.py

```diff
@@ -101,16 +108,18 @@ class TempDirectory(object):
     def __init__(
         self,
         path=None,    # type: Optional[str]
-        delete=None,  # type: Optional[bool]
+        delete=_default,  # type: Union[Optional[bool], _Default]
```

Comment:
`Union[bool, None, _Default]` looks clearer to me.

Reply:
[Done here](https://github.com/pypa/pip/pull/7652/files#diff-bd9cdc0022571f341d8a451e918e279bR111).